webroot/js/utils/chat.js

```diff
@@ -0,0 +1,190 @@
+  let formattedText = new showdown.Converter({
+    emoji: true,
+    openLinksInNewWindow: true,
+    tables: false,
+    simplifiedAutoLink: false,
+    literalMidWordUnderscores: true,
+    strikethrough: true,
+    ghMentions: false,
+  }).makeHtml(message);
+
+  formattedText = linkify(formattedText, message);
+  formattedText = highlightUsername(formattedText, username);
+
+  return addNewlines(formattedText);
+}
+
+function highlightUsername(message, username) {
+	const pattern = new RegExp('@?' + username.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&'), 'gi');
+  return message.replace(pattern, '<span class="highlighted font-bold bg-orange-500">$&</span>');
+}
+
+function linkify(text, rawText) {
+  const urls = getURLs(stripTags(rawText));
+  if (urls) {
+    urls.forEach(function (url) {
+      let linkURL = url;
+
+      // Add http prefix if none exist in the URL so it actually
+      // will work in an anchor tag.
+      if (linkURL.indexOf('http') === -1) {
+        linkURL = 'http://' + linkURL;
+      }
+
+      // Remove the protocol prefix in the display URLs just to make
+      // things look a little nicer.
+      const displayURL = url.replace(/(^\w+:|^)\/\//, '');
+      const link = `<a href="${linkURL}" target="_blank">${displayURL}</a>`;
+      text = text.replace(url, link);
+
+      if (getYoutubeIdFromURL(url)) {
+        if (isTextJustURLs(text, [url, displayURL])) {
+          text = '';
+        } else {
+          text += '<br/>';
+        }
+
+        const youtubeID = getYoutubeIdFromURL(url);
+        text += getYoutubeEmbedFromID(youtubeID);
+      } else if (url.indexOf('instagram.com/p/') > -1) {
+        if (isTextJustURLs(text, [url, displayURL])) {
+          text = '';
+        } else {
+          text += `<br/>`;
+        }
+        text += getInstagramEmbedFromURL(url);
+      } else if (isImage(url)) {
+        if (isTextJustURLs(text, [url, displayURL])) {
+          text = '';
+        } else {
+          text += `<br/>`;
+        }
+        text += getImageForURL(url);
+      }
+    }.bind(this));
+  }
+  return text;
+}
+
+function isTextJustURLs(text, urls) {
+  for (var i = 0; i < urls.length; i++) {
+    const url = urls[i];
+    if (stripTags(text) === url) {
+      return true;
+    }
+  }
+  return false;
+}
+
+
+function stripTags(str) {
+	return str.replace(/<\/?[^>]+(>|$)/g, "");
+}
+
+function getURLs(str) {
+	var exp = /((\w+:\/\/\S+)|(\w+[\.:]\w+\S+))[^\s,\.]/ig;
+	return str.match(exp);
+}
+
+function getYoutubeIdFromURL(url) {
+	try {
+		var regExp = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|\&v=)([^#\&\?]*).*/;
+		var match = url.match(regExp);
+
+		if (match && match[2].length == 11) {
+			return match[2];
+		} else {
+			return null;
+		}
+	} catch (e) {
+		console.log(e);
+		return null;
+	}
+}
+
+function getYoutubeEmbedFromID(id) {
+	return `<iframe class="chat-embed youtube-embed" src="//www.youtube.com/embed/${id}" frameborder="0" allowfullscreen></iframe>`;
```

Comment:
suggestion: You might consider using the `<lite-youtube>` component for this; I've found it to be much faster to load than the standard iframe-based embed.

https://github.com/justinribeiro/lite-youtube